engine/kv_engine.cpp

```diff
@@ -967,9 +975,16 @@ Status KVEngine::HashGetImpl(const StringView& key, std::string* value,
     if (!is_found || (hash_entry.GetRecordType() & DeleteRecordType)) {
       return Status::NotFound;
     }
-
     void* pmem_record = nullptr;
-    if (hash_entry.GetRecordType() & (StringDataRecord | DlistDataRecord)) {
+    if (hash_entry.GetRecordType() == StringDataRecord) {
+      ExpiredTimeType expired_time =
+          hash_entry.GetIndex().string_record->GetExpiredTime();
+      if (expired_time > 0 && expired_time <= TimeUtils::millisecond_time()) {
```

Comment:
We should offer Record::has_expired() interface to do this check.
If user set expired_time to -2, what should we do?

Reply:
move `CheckIsExpired` into utils.hpp